### PR TITLE
Simplify Marshaller::_validate().

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: src/ORM/Marshaller.php [hunk 1]
```diff
@@ -240,32 +240,22 @@ public function one(array $data, array $options = []): EntityInterface
...
      *
      * @param array $data The data to validate.
-     * @param array<string, mixed> $options The options passed to this marshaller.
+     * @param bool|string $validator Validator name. `true` to use the default validator.
      * @param bool $isNew Whether it is a new entity or one to be updated.
      * @return array The list of validation errors.
      * @throws \RuntimeException If no validator can be created.
      */
-    protected function _validate(array $data, array $options, bool $isNew): array
+    protected function _validate(array $data, bool|string $validator, bool $isNew): array
```
Comment (ADmad) [inline src/ORM/Marshaller.php:248]: I would prefer the `Form::validate()` in #15815 also to be simple like this.

File: src/ORM/Marshaller.php [hunk 2]
```diff
@@ -240,32 +240,22 @@ public function one(array $data, array $options = []): EntityInterface
...
 
-        $validator = null;
-        if ($options['validate'] === true) {
-            $validator = $this->_table->getValidator();
-        } elseif (is_string($options['validate'])) {
-            $validator = $this->_table->getValidator($options['validate']);
-        }
-
-        if ($validator === null) {
-            throw new RuntimeException(sprintf(
```
Comment (ADmad) [inline src/ORM/Marshaller.php:262]: Not passing an array allows using argument types and avoids such checks.

File: src/ORM/Marshaller.php [hunk 3]
```diff
@@ -240,32 +239,22 @@ public function one(array $data, array $options = []): EntityInterface
      * Returns the validation errors for a data set based on the passed options
      *
      * @param array $data The data to validate.
-     * @param array<string, mixed> $options The options passed to this marshaller.
+     * @param bool|string $validator Validator name. `true` to use the default validator.
```
Comment (othercorey) [inline src/ORM/Marshaller.php:242]:
```suggestion
     * @param string|bool $validator Validator name or `true` for default validator.
```

File: src/ORM/Marshaller.php (hunk 1 above)
Comment (othercorey) [reply src/ORM/Marshaller.php]: Is false supposed to disable it? If not, should we use `null` to indicate default instead?

Comment (ADmad) [reply src/ORM/Marshaller.php]: > Is false supposed to disable it?

Yes, one can use `'validate' => false` in `$options` for `Table::newEntity()/patchEntity()` to disable validation.

File: src/ORM/Marshaller.php [hunk 4]
```diff
@@ -240,32 +239,22 @@ public function one(array $data, array $options = []): EntityInterface
...
      *
      * @param array $data The data to validate.
-     * @param array<string, mixed> $options The options passed to this marshaller.
+     * @param string|bool $validator Validator name or `true` for default validator.
      * @param bool $isNew Whether it is a new entity or one to be updated.
      * @return array The list of validation errors.
      * @throws \RuntimeException If no validator can be created.
      */
-    protected function _validate(array $data, array $options, bool $isNew): array
+    protected function _validate(array $data, string|bool $validator, bool $isNew): array
```
Comment (markstory) [inline src/ORM/Marshaller.php:247]: Nice that we don't have to do the runtime throw as we can use a native type.

**Pull request merged**